pkg/kubelet/prober/prober_manager.go

```diff
@@ -253,8 +258,14 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 				ready = result == results.Success
 			} else {
 				// The check whether there is a probe which hasn't run yet.
-				_, exists := m.getWorker(podUID, c.Name, readiness)
-				ready = !exists
+				w, exists := m.getWorker(podUID, c.Name, readiness)
+				if exists {
+					// Trigger an immediate run of the readinessProbe to update ready state
+					w.doProbe()
```

Comment:
for the code cleaness, I'd suggest to assign `ready = false` in this branch explicitly. It will be false as a default value, but I'd rather see it being set explicitly